src/cmdstan/command_helper.hpp

```diff
@@ -750,109 +750,28 @@ void check_file_config(argument_parser &parser) {
   }
 }
 
-void init_callbacks(
-    argument_parser &parser,
-    std::vector<stan::callbacks::unique_stream_writer<std::ofstream>>
-        &sample_writers,
-    std::vector<stan::callbacks::unique_stream_writer<std::ofstream>>
-        &diag_csv_writers,
-    std::vector<stan::callbacks::json_writer<std::ofstream>>
-        &diag_json_writers) {
-  auto user_method = parser.arg("method");
-  unsigned int num_chains = get_num_chains(parser);
-  unsigned int id = get_arg_val<int_argument>(parser, "id");
-  int sig_figs = get_arg_val<int_argument>(parser, "output", "sig_figs");
-  bool save_single_paths
-      = user_method->arg("pathfinder")
-        && get_arg_val<bool_argument>(parser, "method", "pathfinder",
-                                      "save_single_paths");
-  std::string output_file
-      = get_arg_val<string_argument>(parser, "output", "file");
-  std::string diagnostic_file
-      = get_arg_val<string_argument>(parser, "output", "diagnostic_file");
-  std::vector<std::string> output_filenames;
-  std::vector<std::string> diagnostic_filenames;
-  sample_writers.reserve(num_chains);
-  diag_csv_writers.reserve(num_chains);
-  diag_json_writers.reserve(num_chains);
-
-  // default - no diagnostics
-  for (int i = 0; i < num_chains; ++i) {
-    diag_csv_writers.emplace_back(nullptr, "# ");
-    diag_json_writers.emplace_back(
-        stan::callbacks::json_writer<std::ofstream>());
+template <typename T>
+void init_null_callbacks(std::vector<T> &writers, size_t num_chains) {
```

Comment:
Glad these functions worked out, I think the result is very nice.

Reply:
Can we unify the naming between them (`callbacks` vs `writers` - I prefer `writers`) -- my bad!